testsuite/tests/contrib/lkt_semantic/array_type/test.lkt

```diff
@@ -0,0 +1,45 @@
+# TODO: support @invalid
+
+val a: Array[Int] = []
+val b: Array[Int] = [1]
+val c: Array[Int] = [1, 2, 3]
+
+# @invalid val a = []
+
+# @invalid val a = [1]
+
+val i: Int = 1
+# TODO: should infer type from i
+# val a = [i, 2, 3]
```

Comment:
It seems that a lot of commented lines are for the same feature (inter array element type inference): I suggest to move unsupported cases to a separate testcase, uncommented and with an expected failure. (commented code is too easy to forget! ;-))

Expected failures (XFAIL tests) allow us to know that a test is failing but still run it to spot when it starts to work in the future. Look for `control` entries in the testsuite’s `test.yaml` files for inspiration, and here is the doc for this mechanism: https://e3-testsuite.readthedocs.io/en/latest/api_control.html